test/test_linalg.py

```diff
@@ -575,6 +575,63 @@ def run_test(upper):
         for upper in [True, False]:
             run_test(upper)
 
+    @skipCUDAIfNoMagmaAndNoCusolver
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_cholesky_with_info(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test(n, batch):
+            A = random_hermitian_pd_matrix(n, *batch, dtype=dtype, device=device)
```

Comment:
I'm curious about these tests because we already have a lot of tests for cholesky, and cholesky_ex and cholesky are "near aliases" of each other; should we just refactor the cholesky tests so that test_cholesky_foo and test_cholestky_foo_ex both call a common helper, but pass a different torch function? Then there can be separate tests that validate their divergent behavior on infos

Reply:
I completely agree. I think it would take a lot of energy from me to refactor the tests now. If those functions were aliases the task of course would be easy. I'd prefer to do it in a follow-up PR, after 1.9 branch cut, if that's okay.